sdk/nodejs/runtime/resource.ts

```diff
@@ -857,8 +857,13 @@ export async function prepareResource(
     // the Resources pointed to by any Dependency objects we encounter, adding them to 'propertyDependencies'.
     const [serializedProps, propertyToDirectDependencies] = await serializeResourceProperties(label, props, {
         // To initially scope the use of this new feature, we only keep output values when
-        // remote is true (for multi-lang components).
+        // remote is true (for multi-lang components, i.e. MLCs).
         keepOutputValues: remote,
+        // When remote is true, exclude resource references from 'propertyDependencies'.
```

Comment:
Why aren't we just removing them always? I don't think any custom resources use resource references, but if they did I still don't think we need to send it in propertyDeps?

Reply:
There are cases like [this](https://github.com/pulumi/pulumi-aws/blob/6ef58a7af22a351f82ebfb8dab00b522495a7eff/sdk/nodejs/s3/bucketObject.ts#L494):

```typescript
export interface BucketObjectArgs {
    bucket: pulumi.Input<string | Bucket>;
}
```

That end up getting sent from the program SDK to the engine as a resource reference when a `Bucket` is passed, and I do think we want to keep the dependency in cases like this.